### PR TITLE
Match filemenu_set_cursor_goal_pos

### PR DESCRIPTION
**Pull request by JoshDuMan**
https://decomp.me/scratch/xnc5w
https://decomp.me/scratch/9QaEy

### CODE REVIEW

File: src/163400.c [hunk 1]
```diff
@@ -81,7 +81,37 @@ void filemenu_set_cursor_alpha(s32 arg0) {
     filemenu_cursorGoalAlpha2 = arg0;
 }
 
-INCLUDE_ASM(s32, "163400", filemenu_set_cursor_goal_pos);
+void filemenu_set_cursor_goal_pos(s32 windowID, s32 posX, s32 posY) {
+    Window* window = &gWindows[windowID];
+
+    if (D_80249BB0 != 0
```
Comment (ethteck) [inline src/163400.c]: I think this can all go on the same line and fit far under 120 chars

File: src/163400.c [hunk 2]
```diff
@@ -81,7 +81,37 @@ void filemenu_set_cursor_alpha(s32 arg0) {
...
+            || get_game_mode() == 0xF
+            || get_game_mode() == 0xD) {
+        if (D_80249BB0 != 0) {
+            s32 i;
+
+            for (i = 0x2C; i < 0x40; i++) {
```
Comment (ethteck) [inline src/163400.c:93]:
```suggestion
            for (i = 0x2C; i < ARRAY_COUNT(gWindows); i++) {
```

File: src/163400.c [hunk 3]
```diff
@@ -81,7 +81,37 @@ void filemenu_set_cursor_alpha(s32 arg0) {
...
+                Window* window = &gWindows[i];
+                s8 parent = window->parent;
+
+                if ((parent == -1 || parent == 0x2C) && (window->flags & 8)) {
+                    break;
+                }
+            }
+            if (i >= 0x40) {
```
Comment (ethteck) [inline src/163400.c:101]:
```suggestion
            if (i >= ARRAY_COUNT(gWindows)) {
```

File: src/163400.c [hunk 4]
```diff
@@ -81,7 +81,37 @@ void filemenu_set_cursor_alpha(s32 arg0) {
...
+                D_80249BB0 = 0;
+            }
+        }
+        D_80249BA0 = posX;
+        D_80249B94 = posX;
+        D_80249BA4 = posY;
+        D_80249B98 = posY;
+
```
Comment (ethteck) [inline src/163400.c:109]: I think this empty line isn't really that useful. If anything I'd add one 4 above before the first `... = posX` statement. but not really important

File: src/163400.c [hunk 5]
```diff
@@ -81,7 +81,37 @@ void filemenu_set_cursor_alpha(s32 arg0) {
     filemenu_cursorGoalAlpha2 = arg0;
 }
 
-INCLUDE_ASM(s32, "163400", filemenu_set_cursor_goal_pos);
+void filemenu_set_cursor_goal_pos(s32 windowID, s32 posX, s32 posY) {
+    Window* window = &gWindows[windowID];
+
+    if (D_80249BB0 != 0
+            || get_game_mode() == 0xF
```
Comment (ethteck) [inline src/163400.c:88]: Do we have a game mode enum? We probably should if we don't

**Review by ethteck — CHANGES_REQUESTED** [review]

File: src/163400.c (hunk 3 above)
Comment (JoshDuMan) [reply src/163400.c:101]: I can do array count here, but it seems like gWindows has subdivisions within it for each range that probably need an enum.

File: src/163400.c (hunk 5 above)
Comment (JoshDuMan) [reply src/163400.c:88]: No we do not, at least from what I saw

Comment (JoshDuMan) [reply src/163400.c:88]: nvm Im wrong

**Comment by BowserSlug** [comment]
🚀 This PR matches 1 function (+0.01%) on `us`.

File: src/163400.c (hunk 3 above)
Comment (ethteck) [reply src/163400.c:101]: yeah or some defines. but I think it's safe to use this for the end of the array. Or it's multiple arrays..hm

**Review by ethteck — APPROVED** [review]

**Review by nanaian — APPROVED** [review]

**Pull request merged**